### PR TITLE
Clarify docs around custom resolvers

### LINKED ISSUE #3524: Clarify docs around custom resolvers
## Description

https://docs.kedro.org/en/stable/configuration/advanced_configuration.html#how-to-use-custom-resolvers-with-omegaconfigloader two problems:

1. for a newbie, it's not clear where the code should go (https://linen-slack.kedro.org/t/16316616/hello-this-is-my-grid-parameter-tuning-options-grid-batch-si#bc370de6-d106-45ef-9cb0-a3a79baa9e09)
2. the code snippet mixes library code with REPL code, which makes it very confusing

### PR DESCRIPTION
**Pull request by ElenaKhaustova**
## Description
Close #3524

## Development notes
Updated `Advanced configuration without a full Kedro project` docs section:

- clarified that code can be run just as a Python script as a full Kedro project is not required;
- split the code snippet from the output for clarity. 

## Developer Certificate of Origin
We need all contributions to comply with the [Developer Certificate of Origin (DCO)](https://developercertificate.org/). All commits must be signed off by including a `Signed-off-by` line in the commit message. [See our wiki for guidance](https://github.com/kedro-org/kedro/wiki/Guidelines-for-contributing-developers/).

If your PR is blocked due to unsigned commits, then you must follow the instructions under "Rebase the branch" on the GitHub Checks page for your PR. This will retroactively add the sign-off to all unsigned commits and allow the DCO check to pass.

## Checklist

- [x] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [x] Signed off each commit with a [Developer Certificate of Origin (DCO)](https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/managing-repository-settings/managing-the-commit-signoff-policy-for-your-repository)
- [x] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [x] Updated the documentation to reflect the code changes
- [x] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [x] Added tests to cover my changes
- [x] Checked if this change will affect Kedro-Viz, and if so, communicated that with the Viz team

### CODE REVIEW

**Review by noklam — APPROVED** [review]

File: docs/source/configuration/advanced_configuration.md [hunk 1]
```diff
@@ -342,17 +343,23 @@ config_loader = OmegaConfigLoader(conf_source=".")
...
 
->>> config_loader["parameters"]
-{'learning_rate': 0.01, 'train_test_ratio': 0.7}
+print(config_loader["parameters"])
+```
+
+If you run it from the same directory where `parameters.yml` placed it gives the following output:
+
+```console
+{'learning_rate': 0.01, 'train_test_ratio': 0.7}}
```
Comment (astrojuanlu) [inline docs/source/configuration/advanced_configuration.md:352]:
```suggestion
{'learning_rate': 0.01, 'train_test_ratio': 0.7}
```

?

**Review by astrojuanlu — APPROVED** [review]
💯 thanks @ElenaKhaustova ! Approved with 1 nitpick

File: docs/source/configuration/advanced_configuration.md (hunk 1 above)
Comment (ElenaKhaustova) [reply docs/source/configuration/advanced_configuration.md:352]: Sure, thank you

**Comment by astrojuanlu** [comment]
The Quora 403 happens from time to time, a restart should drive it away

**Comment by ElenaKhaustova** [comment]
> The Quora 403 happens from time to time, a restart should drive it away

A couple of restarts didn't help, and there's one more issue when fixing `Quora`, so I opened another PR with fixes: #3773

**Review by ankatiyar — APPROVED** [review]
💯

**Pull request merged**